# CSE110 Lab Assignment 1

This assignment is to guide you to solve basic programming problems in Python. Since this is the first assignment, the first problem is solved for you with explanations. Some problems are partially solved to help you. Go through them and try to understand each statment. If you can understand the problems and the solutions done for you, you can solve the rest of the tasks by yourself. 

**<font color='red'>[MUST MAINTAIN VARIABLE NAMING CONVENTIONS FOR ALL THE TASKS]</font>**

## Write the Python code of the following problems:


In [1]:
file=open(file="task 3 input 1.txt",mode="r")
lines=file.read()
print(lines)

I      love         Python          Programming


### Task 1

Write Python code of a program that reads two numbers from the user, and prints their sum, product, and difference.

==========================================================

*hint: Subtract the second number from the first one*

==========================================================

**Example01:**

Input:\
 4\
 5

Output:\
Sum = 9\
Product = 20\
Difference = -1

==========================================================

**Example02:**

Input:\
 30\
 2

Output:\
Sum = 32\
Product = 60\
Difference = 28

==========================================================

For your explanation, the first question's code is done below. Try relating the block of code with the lesson you have learned and understand the significance of each line. You're most welcome to play around with the code. This will strengthen your understanding.


In [ ]:
### Take input of 2 numbers from the user
var_1 = input('Please Enter Your First Number')
var_2 = input('Please Enter Your Second Number')

#Since input() function converts everything to String,
#for performing any kind of mathematical operation you need to convert them to int.
#For this conversion, we need to use int() function


# First, let's clarify whether the inputs are actually Strings or not. 
print(type(var_1))
print(type(var_2))


# Convert Strings to integer using the int() function
var_3 = int(var_1)
var_4 = int(var_2)

#===============================================================
#input taking and conversion can be done in a single sentense
#var_1 = int(input('Please Enter Your First Number'))
#var_2 = int(input('Please Enter Your Second Number'))

#===============================================================

# Perform Addition
sum = var_3 + var_4

# Perform Multiplication 
product = var_3 * var_4

# Perform Substraction 
difference = var_3 - var_4

# Print all the calculated results
print("Sum =", sum)
print("Product =", product)
print("Difference =", difference)

Please Enter Your First Number30
Please Enter Your Second Number2
<class 'str'>
<class 'str'>
Sum = 32
Product = 60
Difference = 28


### Task 2

Write Python code of a program that reads the radius of a circle and prints its circumference and area.

==========================================================

*hint(1): import math and then use **math.pi** for getting the value of pi.\
For details read from https://docs.python.org/3/library/math.html*

*hints(2): You can import math and use math function for making squares math.pow(number, power)   Or you can simply write using power opeartor. For example:  S**2.*

==========================================================

**Example01:**

Input: 4

Output:\
Area is 50.26548245743669\
Circumference is 25.132741228718345

==========================================================

**Example02:**

Input: 3.5

Output:\
Area is 38.48451000647496\
Circumference is 21.991148575128552

==========================================================

For your explanation, the the first part of the question(area calculation) is done below.

In [1]:
import math 


radius  = float(input("please enter the radius value:")) 

area = math.pi * radius **2 
print("Area result is:", area)

# area = math.pi * math.pow(radius, 2)
# print("Area result is:", area)

# area = math.pi * radius * radius
# print("Area result is:", area)

# TODO
# calculate circumference

please enter the radius value:10.5
Area result is: 346.3605900582747


### Task 3

Write Python code of a program that reads two numbers from the user. Your program should then print "First is greater" if the first number is greater, "Second is greater" if the second number is greater, and "The numbers are equal" otherwise.

==========================================================

**Example:**

Input:\
-4\
-4

Output:\
The numbers are equal

==========================================================

**Example:**

Input:\
-40\
-4

Output:\
Second is greater



In [2]:
#Todo
first_number=int(input("Enter the number: "))
second_number=int(input("Enter the number: "))
if(first_number>second_number):
    print("First is greater")
elif(first_number<second_number):
    print("Second is greater")
else:
    print("The numbers are equal")

Enter the number: -4
Enter the number: -4
The numbers are equal


### Task 4

Write Python code of a program that reads two numbers, subtracts the smaller number from the larger one, and prints the result.

*hint(1): First check which number is greater*

==========================================================

**Example:**

Input:\
-40\
-4

Output:\
36

==========================================================

**Example:**

Input:\
6\
2

Output:\
4


In [ ]:
#Todo
num1=int(input("enter the first number: "))
num2=int(input("enter the second number: "))

if(num1>num2):
    subtracts= num1 - num2
    print(" ",subtracts)
elif(num1<num2):
    subtracts= num2 - num1
    print(" ",subtracts)
else:
    print("Both qual")

### Task 5

Write Python code of a program that reads a number, and prints "The number is even" or "The number is odd", depending on whether the number is even or odd. 

*hint(1): use the modulus (%) operator*

*hint(2): You can consider the number to be an integer* 

==========================================================

**Example:**

Input:\
5

Output:\
The number is odd


==========================================================

**Example:**

Input:\
-44

Output:\
The number is even


In [ ]:
#Todo
fnumber=int(input('Enter the number: '))
if(fnumber%2==0):
    print("The number is even")
else:
    print("The number is odd")

### Task 6

Write Python code of a program that reads an integer, and prints the integer if it is a multiple of **either 2 or 5.**

For example, 2, 4, 5, 6, 8, 10, 12, 14, 15, 16, 18, 20, 22 …

==========================================================

*hint(1): use the modulus (%) operator for checking the divisibility*

*hint(2): You can consider the number to be an integer* 


==========================================================

**Example01:**

Input:\
5

Output:\
5


==========================================================

**Example02:**

Input:\
10

Output:\
10

==========================================================

**Example03:**

Input:\
3

Output:\
Not a multiple


In [ ]:
#Todo
number=int(input("Enter the number: "))
if(number%2==0 or number%5==0):
    print("",number)
else:
    print("Not a multiple")

In [2]:
number=int(input(""))
# if number%2:
#     print(number)
# elif number%5==0:
#     print(number)


if number %2==0 or number%5==0:
    print(number)
else:
    print("amar int value ta valid na ")
    

3
amar int value ta valid na 


### Task 7
Write Python code of a program that reads an integer, and prints the integer  it is a multiple of **either 2 or 5 but not both.** 

For example, 2, 4, 5, 6, 8, 12, 14, 15, 16, 18, 22 …

==========================================================

*hint(1): use the modulus (%) operator for checking the divisibility*

*hint(2): You can consider the number to be an integer* 

==========================================================

**Example01:**

Input:\
5

Output:\
5

==========================================================

**Example02:**

Input:\
10

Output:\
multiple of 2 and 5 both

==========================================================

**Example02:**

Input:\
44

Output:\
44

In [2]:
#Todo
number=int(input("Enter the number: "))
if(number%2==0 and number%5==0):
    print("multiple of 2 and 5 both")
elif(number%2==0):
    print(" ",number)
elif(number%5==0):
    print(" ",number)
else:
    print("None of this")

Enter the number: 3
None of this


### Task 8
Write Python code of a program that reads an integer, and prints the integer if it is a multiple of **2 and 5.**


==========================================================

For example, 10, 20, 30, 40, 50 …

*hint(1): use the modulus (%) operator for checking the divisibility*

*hint(2): You can consider the number to be an integer* 


==========================================================

**Example01:**

Input:\
5

Output:\
Not multiple of 2 and 5 both


==========================================================

**Example02:**

Input:\
40

Output:\
40

==========================================================

**Example02:**

Input:\
40

Output:\
40

In [ ]:
#Todo
number=int(input("enter the number: "))
if(number%2==0 and number%5==0):
    print(" ",number)
else:
    print("Not multiple of 2 and 5 both")

### Task 9


Write Python code of a program that reads an integer, and prints the integer if it is a multiple of **NEITHER 2 NOR 5.** 

For example, 1, 3, 7, 9, 11, 13, 17, 19, 21, 23, 27, 29, 31, 33, 37, 39 …

==========================================================

*hint(1): use the modulus (%) operator for checking the divisibility*

*hint(2): You can consider the number to be an integer* 


==========================================================

**Example01:**

Input:\
3

Output:\
3


==========================================================

**Example02:**

Input:\
19

Output:\
19

==========================================================

**Example03:**

Input:\
5

Output:\
No

==========================================================

**Example04:**

Input:\
12

Output:\
No


In [8]:
#Todo
number=int(input("Enter the number: "))
if(number%2!=0 and number%5!=0):
    print(" ",number)
else:
    print("No")

Enter the number: 19
  19


### Task 10

Write Python code of a program that reads an integer, and prints the integer if it is **NOT a multiple of 2** OR **NOT a multiple of 5.**

For example, 1, 2, 3, 4, 5, 6, 7, 8, 9, 11, 12, 13, 14, 15, 16, 17, 18, 19, 21, 22

==========================================================

*hint(1): use the modulus (%) operator for checking the divisibility*

*hint(2): You can consider the number to be an integer* 


==========================================================

**Example01:**

Input:\
3

Output:\
3

==========================================================

**Example02:**

Input:\
11

Output:\
11

==========================================================

**Example03:**

Input:\
20

Output:\
No

In [6]:
number=int(input("Enter the number: "))
if(not(number%2==0 or number%5==0)):
    print(" ", number)
else:
    print("No")

Enter the number: 20
No


### Task 11

Write Python code of a program that reads a student’s mark for a single subject, and prints out the corresponding grade for that mark. The mark ranges and corresponding grades are shown in the table below. You need to make sure that the marks are valid. For example, a student cannot receive -5 or 110. So the valid marks range is 0 to 100.

*hint(1): You can consider the number to be an integer* 

*hint(2): This problem can be solved in two ways: top-down (starts from A) and bottom-up (starts from F)*


| Marks | Grage|
| --- | --- | 
|90 or above| A |
|  |  | 
|80-89| B |
|  |  | 
|70-79| C |
|  |  | 
|60-69| D |
|  |  | 
|50-59| E |
|  |  | 
|Below 50| F |







In [34]:
#Todo
mark=int(input("Enter the a student’s mark's "))
if(mark>=0 and mark<=100):
    
    if(mark<50):
        print("F")
    elif(mark>=50 and mark<=59):
        print("E")
    elif(mark>=60 and mark<=69):
         print("D")
    elif(mark>=70 and mark<=79):
         print("C")
    elif(mark>=80 and mark<=89):
         print("B")
    elif(mark>=90):
        print("A")
else:
    print("NO valid")


Enter the mark-2
NO valid


In [8]:
Hour =int(input(""))
print((60*300)+40*200)

100
26000


### Task 12

Write Python code of a program to compute and display a person’s weekly salary as determined by the following conditions:
If the hours worked are less than or equal to 40, the person receives Tk200.00 per hour, else the person receives Tk8000.00 plus Tk300.00 for each hour worked over 40 hours.
The program should request the hours worked as input and should display the salary as output.

==========================================================

*hint: You can consider the hour(the user input) to be an integer* 

==========================================================

**Example1:**\
Input: 100\
Output: 26000

==========================================================

**Example2:**\
Input: 30\
Output: 6000


In [39]:
#Todo#Todo
Hour=int(input("Enter the hour"))
if(Hour<=40):
    salary=Hour*200
    print("salary= ", salary)
else:
    salary=(8000+(Hour-40)*300)
    print("salary= ",salary)

Enter the hour45
salary=  9500


### Task 13
Write Python code of a program that finds the number of hours, minutes, and seconds in a **given number of seconds.** 

==========================================================

*hint(1): The user input will be an integer value* 

*hint(2): 1 hour = 60 mins = 3600 seconds*\
*1min = 60 seconds* 

==========================================================


**Example01:**\
Input: 10000\
Output: Hours: 2  Minutes: 46  Seconds: 40

==========================================================

**Example02:**\
Input: 500\
Output: Hours: 0  Minutes: 8  Seconds: 20




In [43]:
seconds=int(input("Enter the given number "))
Hour=seconds//3600
left_seconds=seconds%3600
mins=left_seconds//60
seconds=left_seconds%60
print( "Hours:",Hour, "Minutes:",mins, "Seconds:",seconds)

Enter the given number 500
Hours: 0 Minutes: 8 Seconds: 20


In [12]:
print(5//2)

2


In [13]:
s=int(input(""))
if s<100:
    print(3000-125*(s**2))
elif s>=100:
    print(12000/(4+(s**2)/14900))

120
2416.2162162162163


### Task 14
Suppose the following expressions are used to calculate the values of L for different values of S:

$L =  3000-125S^2$ 
if     $S<100$

$ L =  \frac{12000}{4+S^2/14900}$
if     $S\geq100$

Write a Python code of a program that reads a value of S and then calculates the value of L.

==========================================================

*hint(1): You can import math and use math function for making squares math.pow(number, power)   Or you can simply write S**2.*

*hint(2): The value of S(user input) will be an integer* 

==========================================================

**Example01:**\
Input: 120\
Output: 2416.2162162162163


==========================================================

**Example02:**\
Input: 3\
Output: 1875


In [4]:
S=int(input("Enter the value: "))
if(S<100):
    L=3000-125*S**2
    print(" ",L)
else:
    (S>=100)
    L=12000/(4+S**2/14900)
    print(" ",L)

Enter the value: 120
  2416.2162162162163


### Task 15

Take an hour from the user as input and tell it is time for which meal.

•	The user will input the number in a 24-hour format. So, 14 means 2 pm, 3 means 3 am, 18 means 6 pm, etc.\
•	Valid inputs are 0 to 23. Inputs less than 0 or more than 23 are invalid in 24-hour clock.\
•	Assume, Input will be whole numbers. For example, 3.5 will NOT be given as input.

==========================================================

Inputs: Message to be printed\
4 to 6: Breakfast\
12 to 13: Lunch\
16 to 17: Snacks\
19 to 20: Dinner\
For all other valid inputs, say "Patience is a virtue"\
For all other invalid inputs, say "Wrong time"

==========================================================

For example,\
If the user enters 4, your program should print the message "Breakfast".\
If the user enters 5, your program should print the message "Breakfast".\
If the user enters 6, your program should print the message "Breakfast".\
If the user enters 0, your program should print the message "Patience is a virtue".\
If the user enters 1, your program should print the message "Patience is a virtue".\
If the user enters 18, your program should print the message "Patience is a virtue".\
If the user enters 23, your program should print the message "Patience is a virtue".\
If the user enters 24, your program should print the message "Wrong Time".\
If the user enters -1, your program should print the message "Wrong Time".\
If the user enters 27, your program should print the message "Wrong time".

==========================================================

Hints:\
You can use nested conditionals (if-else) or chained conditions (if-elif-else) to solve this problem.



In [ ]:
#Todo
hour=int(input("Enter the hour"))
if(hour>=0 and hour<=23):
    if(hour>=4 and hour<=6):
        print("Breakfast")
    elif(hour>=12 and hour<=13):
        print("Lunch")
    elif(hour>=16 and hour<=17):
        print("snacks")
    elif(hour>=19 and hour<=20):
        print("Dinar")
    else:
        print("Patience is a virtue")
else:
    print("Wrong time")